src/interpreter.rs

```diff
@@ -499,9 +499,9 @@ pub fn eval_script(
                         OP_1NEGATE | OP_1 | OP_2 | OP_3 | OP_4 | OP_5 | OP_6 | OP_7 | OP_8
                         | OP_9 | OP_10 | OP_11 | OP_12 | OP_13 | OP_14 | OP_15 | OP_16 => {
                             // ( -- value)
-                            let bn =
-                                ScriptNum(i64::from(u8::from(pv)) - i64::from(u8::from(OP_1) - 1));
-                            stack.push_back(bn.getvch());
+                            let bn = ScriptNum::from(u8::from(pv))
+                                - (ScriptNum::from(u8::from(OP_1) - 1));
```

Comment:
Is there a not already a constant that corresponds to `ScriptNum::from(u8::from(OP_1) - 1)`?

Reply:
Yeah – `OP_RESERVED`. I kept it like this to mirror the C++, but I would be _more_ than happy to change that.

I assume it’s done this way because `OP_RESERVED` isn’t obviously in the position that an `OP_0` would be (if it wasn’t already somewhere else).